Jenkinsfile

```diff
@@ -38,12 +46,20 @@ pipeline {
             }
           }
           steps {
-            echo "Installing Insights..."
-            sh '/bin/python36 -m pip install --user -e .[develop]'
             echo "Testing with Pytest..."
-            sh '/bin/python36 -m pytest'
-            echo "Testing with flake8..."   
-            sh '/bin/python36 -m flake8'
+            sh """
+                /bin/python36 -m venv .testenv
+                source .testenv/bin/activate
+                pip install -e .[testing]
+                pytest
+            """
+            echo "Testing with flake8..."
+            sh """
+                /bin/python36 -m venv .f8env
+                source .f8env/bin/activate
+                pip install -e .[flake8]
```

Comment:
Just wondering, what is the reasoning behind creating a separate environment for _flake8_ and for _pytest_. I understand and favor splitting the two package groups, but what are the benefits of creating two virtualenvs, each with only one of those?

Reply:
@Glutexo The primary reason for multiple virtualenvs is that its the only way to ensure the isolated set of python modules is installed for that part of the run.  If we combine `pytest` and `flake8` then we could just use 1 section with a single virtualenv.

Reply:
You’re right. This way you don’t only test the actual _tests_ themselves, but the environments too. Okay. Thanks. 👍